src/models/PrimaryAuth.js

```diff
@@ -106,12 +106,21 @@ function (Okta, BaseLoginModel, CookieUtil, Enums) {
 
       var primaryAuthPromise;
 
-      if (this.appState.get('isUnauthenticated') && !this.settings.get('features.passwordlessAuth')) {
-        primaryAuthPromise = this.doTransaction(function (transaction) {
-          var authClient = this.appState.settings.authClient;
-          return this.doPrimaryAuth(authClient, signInArgs, transaction.authenticate);
-        });
-      } else {
+      var authClient = this.appState.settings.authClient;
+
+      if (this.appState.get('isUnauthenticated')) {
+        // bootstrapped with stateToken
+        if (!this.settings.get('features.passwordlessAuth')) {
+          primaryAuthPromise = this.doTransaction(function (transaction) {
+            var authClient = this.appState.settings.authClient;
+            return this.doPrimaryAuth(authClient, signInArgs, transaction.authenticate);
+          });
+        } else {
+          primaryAuthPromise = this.performTransaction(authClient, signInArgs);
+        }
+      }
+      else {
+        //normal username/password flow
         primaryAuthPromise = this.startTransaction(function (authClient) {
```

Comment:
I am not sure if we need this else here, but I am leaving it here for now.

Reply:
Looks like this flow is hit for login flow when widget is not bootstrapped with a stateToken.

Reply:
we need this when there is no `stateToken` in the widget bootstrap config (at that point in time we don't have any response from the api yet, so we need this as the default login)